referenceframe/frame.go

```diff
@@ -31,14 +31,14 @@ type Limit struct {
 
 // RestrictedRandomFrameInputs will produce a list of valid, in-bounds inputs for the frame, restricting the range to
 // `lim` percent of the limits.
-func RestrictedRandomFrameInputs(m Frame, rSeed *rand.Rand, lim float64) []Input {
+func RestrictedRandomFrameInputs(m Frame, rSeed *rand.Rand, lim float64, nodePose []Input) []Input {
 	if rSeed == nil {
 		//nolint:gosec
 		rSeed = rand.New(rand.NewSource(1))
 	}
 	dof := m.DoF()
```

Comment:
Should I do error checking here to make sure nodePose has same length as dof

Reply:
Yes.